tsdb/engine/bz1/bz1.go

```diff
@@ -489,13 +489,17 @@ func (e *Engine) writeBlocks(bkt *bolt.Bucket, a [][]byte) error {
 			// Encode block in the following format:
 			//   tmax int64
 			//   data []byte (snappy compressed)
-			value := append(u64tob(uint64(tmax)), snappy.Encode(nil, block)...)
+			buf := make([]byte, 16+snappy.MaxEncodedLen(len(block)))
+			binary.BigEndian.PutUint64(buf[0:8], uint64(tmin))
```

Comment:
Nope. `tmin` is in `buf[0:8]` and the old "`value`" (`tmax` followed by `block`) is in `buf[8:]`. Then at line 499 (new number), you can see that the key is `buf[:8]` (the old `tmin`), and the value is `buf[8:]` (the old `value`)